crypto/rsa/rsa_backend.c

```diff
@@ -96,16 +109,48 @@ int ossl_rsa_fromdata(RSA *rsa, const OSSL_PARAM params[], int include_private)
                                 ossl_rsa_mp_coeff_names))
             goto err;
 
-        /* It's ok if this private key just has n, e and d */
-        if (sk_BIGNUM_num(factors) != 0
-            && !ossl_rsa_set0_all_params(rsa, factors, exps, coeffs))
-            goto err;
-    }
+        if (derive_from_pq && (sk_BIGNUM_num(exps) == 0) && (sk_BIGNUM_num(coeffs) == 0)) {
```

Comment:
The multiprime code will not work inside the fips module... If it gets to the else in FIPS mode it is time to return an error.
So you probably need sections of
#ifndef FIPS_MODULE

Reply:
Note you can see what files are included in the fips module in providers/fips.module.sources